security/cmd/chiron/main.go

```diff
@@ -0,0 +1,183 @@
+				return
+			}
+			runWebhookController()
+		},
+	}
+)
+
+type cliOptions struct {
+	// The namespace of the webhook certificates
+	certificateNamespace string
+	kubeConfigFile       string
+	// The names of the MutatingWebhookConfiguration to manage
+	// In prototype, only one is supported.
+	mutatingWebhookConfigNames string
+	// The names of the webhooks in webhook configurations
+	// In prototype, only one is supported.
+	mutatingWebhookNames string
+
+	// TODO (lei-tang): Add the name of the ValidatingWebhookConfiguration to manage
+
+	// The minimum grace period for cert rotation.
+	certMinGracePeriod time.Duration
+
+	// Monitoring port number
+	monitoringPort int
+
+	logOptions *log.Options
+	// Currently, no topic is registered for ctrlz yet
+	ctrlzOptions *ctrlz.Options
+
+	// The length of certificate rotation grace period, configured as the ratio of the certificate TTL.
+	// If certGracePeriodRatio is 0.2, and cert TTL is 24 hours, then the rotation will happen
+	// after 24*(1-0.2) hours since the cert is issued.
+	certGracePeriodRatio float32
+
+	// Enable profiling in monitoring
+	enableProfiling bool
+
+	// Whether enable the webhook controller
+	enableController bool
+}
+
+func init() {
+	flags := rootCmd.Flags()
+
+	flags.BoolVar(&opts.enableController, "enable-controller", false, "Specifies whether enabling "+
+		"Istio Webhook Controller.")
+	flags.StringVar(&opts.certificateNamespace, "certificate-namespace", "istio-system",
+		"The namespace of the webhook certificates.")
+
+	flags.StringVar(&opts.kubeConfigFile, "kube-config", "",
+		"Specifies path to kubeconfig file. This must be specified when not running inside a Kubernetes pod.")
+
+	// Monitoring configuration
+	flags.IntVar(&opts.monitoringPort, "monitoring-port", 15021, "The port number for monitoring Chiron. "+
+		"If unspecified, Chiron will disable monitoring.")
+	flags.BoolVar(&opts.enableProfiling, "enable-profiling", false, "Enabling profiling when monitoring Chiron.")
+
+	// Certificate issuance configuration.
+	flags.Float32Var(&opts.certGracePeriodRatio, "cert-grace-period-ratio",
+		cmd.DefaultWorkloadCertGracePeriodRatio, "The certificate rotation grace period, as a ratio of the "+
+			"certificate TTL.")
+	flags.DurationVar(&opts.certMinGracePeriod, "cert-min-grace-period",
+		cmd.DefaultWorkloadMinCertGracePeriod, "The minimum certificate rotation grace period.")
+
+	// MutatingWebhook configuration
+	flags.StringVar(&opts.mutatingWebhookConfigNames, "mutating-webhook-config-names", "istio-sidecar-injector",
+		"The names of the mutatingwebhookconfiguration resources in Kubernetes, separated by comma. Chiron will manage them.")
+	flags.StringVar(&opts.mutatingWebhookNames, "mutating-webhook-names", "sidecar-injector.istio.io",
+		"The names of the webhook in mutatingwebhookconfigurations, separated by comma. Only the specified webhooks will be managed by Chiron.")
+
+	// Hide the command line options for the prototype
```

Comment:
why

Reply:
The command line options are marked as hidden as they are experimental.